frontend/public/components/dashboards-page/overview-dashboard/top-consumers-card.tsx

```diff
@@ -84,23 +85,26 @@ const TopConsumersCard_ = connectToURLs(MonitoringRoutes.Prometheus)(({
   stopWatchK8sResource,
   resources,
   urls,
+  flags,
 }: TopConsumersCardProps) => {
   const [type, setType] = React.useState(PODS);
   const [sortOption, setSortOption] = React.useState(MetricType.CPU);
 
   React.useEffect(() => {
-    const topConsumersMap = getTopConsumersQueries();
+    const topConsumersMap = getTopConsumersQueries(flags);
     const currentQuery = topConsumersMap[type].queries[sortOption];
     watchPrometheus(currentQuery);
+
     const k8sResource = getResourceToWatch(topConsumersMap[type].model);
     watchK8sResource(k8sResource);
+
     return () => {
       stopWatchPrometheusQuery(currentQuery);
       stopWatchK8sResource(k8sResource);
     };
-  }, [watchPrometheus, stopWatchPrometheusQuery, watchK8sResource, stopWatchK8sResource, type, sortOption]);
+  }, [watchPrometheus, stopWatchPrometheusQuery, watchK8sResource, stopWatchK8sResource, type, sortOption, flags]);
```

Comment:
Does adding `flags` as a dependency work here since it's not a deep equals? (I have the same question for the other hooks.)

Reply:
@rawagner addressed this issue in commit 2ddc5956efd63629c3189b376c83867cacc14f09 by using `useEffectDeepCompare` hook, which was meant to be part of #2269.

I'm no React hooks expert, but it seems that `useEffectDeepCompare` supports both standard `React.useEffect` comparison behavior (`dependencies` argument) and Lodash `_.isEqual` comparison behavior (`deepCompareDependencies` argument).

Reply:
@spadgett If the `connectToFlags` HOC would use `stateToProps` implementation which treated `flags` not as object, but as a memoized (change driven by `desiredFlags`) **immutable** map, we wouldn't have to use deep comparison, I think.

However, lots of code depends on `connectToFlags` and assumes that `flags` is an object of type `{[key: string]: boolean}`.

Reply:
Maybe I'm misunderstanding, but how do we memoize the result if the flag values change?

In other places, we list each flag separately in the dependencies, but not possible here as the list is dynamic.

Does a simple `JSON.stringify` work? I think we generally want to avoid deep comparisons in general with `useEffect`, so I'm a little hesitant to add a `useEffectDeepCompare`

Maybe @christianvogt has an idea

Reply:
Seems a bit hacky to me. Will `useEffectDeepCompare` only exist until #2296 is complete?

Reply:
@christianvogt  `useEffectDeepCompare` is also used in #2269  ie https://github.com/openshift/console/pull/2269/files#diff-e7411a860eed1cba03ac0052f5367480R70

Reply:
I didn't look at the code closely of the other PR. Thanks for the reference.

I saw some cases where it seems like breaking down the component using this effect into smaller components where each component manages a single watch would eliminate the use of the deep compare and make the components also more efficient.

Reply:
Yes, i guess we could do that but than we give a lot of control over the visuals/logic of contribution and that is something we are trying to avoid AFAIK.  We could take a look at the for 4.3 @vojtechszocs 

We could also use `JSON.stringify` for now, instead of `useEffectDeepCompare`, that should work.
The other thing, hack, we could do for now is to do

```javascript
React.useEffect(() => {
...
}, [..._.values(flags)]);
```

which wont satisfy `eslint(react-hooks/exhaustive-deps)` but will work

Reply:
I'll remove the `useEffectDeepCompare` hook and use the suggested `JSON.stringify` approach for now.

@christianvogt @rawagner Check out [this comment](https://github.com/facebook/react/issues/14476#issuecomment-471199055), it shows several ways to address custom (i.e. deep) comparisons for hook's dependencies. (Thanks @spadgett for sharing this.)

The `flags` object has type `{[key: string]: boolean}` and is therefore shallow, without cycles and fully serializable. The performance cost of `JSON.stringify(flags)` is relatively small.

As for #2269 the `useEffectDeepCompare` hook shouldn't be necessary; if we need custom comparisons in hook's dependencies, we should rethink our approach and adapt according to above.